services/ResourceService/src/index.php

```diff
@@ -0,0 +1,246 @@
+  } 
+  else {
+    // If $id is empty then assume we are dealing with fedora base rest endpoint
+    return $app['config']['islandora']['fedoraProtocol'].'://'.$app['config']['islandora']['fedoraHost'].$app['config']['islandora']['fedoraPath'];
+  }
+};
+
+/** 
+ * Apply middleware to all controllers
+*/
+$app['controllers']
+    ->convert('id', $idToUri)
+    ->assert('id',$app['config']['islandora']['resourceIdRegex'])
+    ->before($htmlHeaderToTurtle) 
+    ->before($hostHeaderNormalize)
+;
+
+/**
+ * Convert returned Guzzle responses to Symfony responses.
+ */
+$app->view(function (ResponseInterface $psr7) {
+  return new Response($psr7->getBody(), $psr7->getStatusCode(), $psr7->getHeaders());
+});
+
+/**
+ * Resource GET route. takes $id (valid UUID or empty) as first value to match, optional a child resource path
+ * takes 'rx' and/or 'metadata' as optional query arguments
+ * @see https://wiki.duraspace.org/display/FEDORA40/RESTful+HTTP+API#RESTfulHTTPAPI-GETRetrievethecontentoftheresource
+ */
+$app->get("/islandora/resource/{id}/{child}",function (Application $app, Request $request, $id, $child) {
+   $tx = $request->query->get('tx', "");
+   $metadata = $request->query->get('metadata', FALSE) ? '/fcr:metadata' : ""; 
+   try {
+     $response = $app['api']->getResource($app->escape($id).'/'.$child.$metadata, $request->headers->all(), $tx);
+   }
+   catch (\Exception $e) {
+     $app->abort(503, 'Chullo says "Fedora4 Repository Not available"');
+   }
+   return $response;
+})
+->value('id',"")
+->value('child',"");
+
+/**
+ * Resource POST route. takes $id (valid UUID or empty) for the parent resource as first value to match
+ * takes 'rx' and/or 'checksum' as optional query arguments
+ * @see https://wiki.duraspace.org/display/FEDORA40/RESTful+HTTP+API#RESTfulHTTPAPI-BluePOSTCreatenewresourceswithinaLDPcontainer
+ */
+$app->post("/islandora/resource/{id}",function (Application $app, Request $request, $id) {
+   $tx = $request->query->get('tx', "");
+   $checksum = $request->query->get('checksum', "");
+   try {
+     $response = $app['api']->createResource($app->escape($id), $request->getContent(), $request->headers->all(), $tx, $checksum);
+   }
+   catch (\Exception $e) {
+     $app->abort(503, '"Chullo says Fedora4 Repository is Not available"');
+   }
+   return $response;
+})
+->value('id',"");
+
+/**
+ * Resource PUT route. takes $id (valid UUID or empty) for the resource to be update/created as first value to match, 
+ * optional a Child resource relative path
+ * takes 'rx' and/or 'checksum' as optional query arguments
+ * @see https://wiki.duraspace.org/display/FEDORA40/RESTful+HTTP+API#RESTfulHTTPAPI-YellowPUTCreatearesourcewithaspecifiedpath,orreplacethetriplesassociatedwitharesourcewiththetriplesprovidedintherequestbody.
+ */
+$app->put("/islandora/resource/{id}/{child}",function (Application $app, Request $request, $id, $child) {
```

Comment:
Hi, child is optional, set to "" by default at https://github.com/Islandora-CLAW/CLAW/pull/131/files#diff-079c3800368dc100b90e9a9254e6fcc6R176 , so if not passed/used in the call it's all good, the route will match. Only post is {$id} only, and i would love to add $child there too